### DISCUSSION TITLE
mydumper exit codes

### DISCUSSION THREAD

**Discussion by UnixRonin** [mydumper/mydumper#1296] [General]
I'm finding mydumper exiting with non-zero return codes after apparently successful dumps, and it's causing my DB backup scripts to fail.  How can I find out what different numeric mydumper exit codes mean so that I can figure out what — if anything — is wrong?  I CAN tell my scripts to just ignore the return code, but for obvious reasons that seems like a bad idea.

**Comment by davidducos** [comment]
Hi @UnixRonin, 
If mydumper is exiting with a non-zero code, then it might be something that it is 100% ok, which doesn't mean that the backup is inconsistent, it could mean that something didn't end as expected. Do you have the log ?

**Reply by UnixRonin** [reply]
I do.  [Here's an example log.](https://bpa.st/IU7IS)  This one exited with result code 14.
(wgetpaste because I can't get github to accept it as a gzipped attachment.)

mydumper command was:
mydumper --defaults-file=/root/.my.cnf -CceGER --trx-consistency-only --threads 4 -o <destination> -l 300 -v 3

Running against MariaDB 10.6.14 in a three-node Galera cluster

Edited to add:  The PREVIOUS invocation, from which I do not have a log, returned a code in the 34000 range as best I recall and was definitely failed, it aborted about 580MB into a 1.7GB dump.

**Reply by UnixRonin** [reply]
And today, with no changes in the script or the database, it ran without errors.  I've been through the log file I pasted yesterday line by line a dozen times and I cannot see anything in the log that looks like an error or even a warning.  Is there any way at all that I can determine what a given error code means, short of going through the source code, or even figure out which non-zero return codes are safe to ignore?

**Reply by ghnp5** [reply]
@davidducos Hey! Is there a list of Return/Exit Codes and what they mean?

I'm having the same problem, and I found this Discussion while trying to find a list of exit codes.

In my case, there was a warning due to deadlock, but it SEEMS that it was able to successfully retry the failed statement.

However, RC = -1, so my script saw this "myloader" run as Failed.

> ** (myloader:1): WARNING **: 09:39:50.978: Thread 17: Error restoring 1: Deadlock found when trying to get lock; try restarting transaction
> ** (myloader:1): WARNING **: 09:39:50.978: Thread 17: Retrying last failed executed statement
> ** Message: 09:39:50.999: Writing resume.partial file
> ** Message: 09:39:51.013: Errors found:
> \- Tablespace:   0
> \- Schema:       0
> \- Data:         0
> \- View:         0
> \- Sequence:     0
> \- Index:        0
> \- Trigger:      0
> \- Constraint:   0
> \- Post:         0
> Retries:        1

**Reply by davidducos** [reply]
Hi @UnixRonin,
This:
`I do. [Here's an example log.](https://bpa.st/IU7IS) This one exited with result code 14.`
It is quite strange, as there is no ERROR or WARNING on the log. I will have to review the code to found where it might be a bug.

@ghnp5, there is no list of codes, as we are counting the amount of errors. Anything more that 1 means that something didn't work as expected. In you rcase, there was a deadlock and myloader retries and succeeded on the second attempt.

**Reply by ghnp5** [reply]
@davidducos - Thank you very much!

In that case, if RC <= 0, I should just consider successful, then? :-)

(so, "-1" means there was no error)

**Reply by davidducos** [reply]
Oh! RC was -1!?!?!? that is not ok, that is a bug. I think that I know where it might be.

**Reply by ghnp5** [reply]
Thank you for raising the Issue and fixing it.

Can I just confirm with you if this was supposed to return RC = 0?
Or was it intended to return RC > 0?

Just so I know what to expect once the issue is fixed.

Thanks again!

**Reply by UnixRonin** [reply]
So @davidducos a POSITIVE exit code means that it should have completed OK but possibly had to retry, but a NEGATIVE exit code means a failure?

**Reply by davidducos** [reply]
@ghnp5 I will need to review the code to determine what is the best on those scenarios. 

@UnixRonin Positive exit code means that something needs to be reviewed to determine if the backup is ok. A negative exit code is a bug

**Reply by UnixRonin** [reply]
Failed again this morning;  status 35584 and return value 139, dumped 1.1MB of 1.8G

**Reply by UnixRonin** [reply]
@davidducos related thought while trying to isolate the issue:

** (mydumper:6000): WARNING **: 11:10:44.660: Using trx_consistency_only, binlog coordinates will not be accurate if you are writing to non transactional tables.

Is there any way to tell mydumper something like "Suppress this warning, I don't HAVE any non-transactional tables"?
~~(This warning APPEARS to result in an exit code 139.)~~   actually, ignore that...
